src/api-service/__app__/onefuzzlib/pools.py

```diff
@@ -1146,21 +1146,26 @@ def update_nodes(self) -> None:
             self.nodes.append(node_state)
 
     def update_configs(self) -> None:
-        if self.state != ScalesetState.running:
-            logging.debug(
-                "scaleset not running, not updating configs: %s", self.scaleset_id
-            )
-            return
+        if not self.needs_config_update:
+            logging.debug("config update not needed: %s", self.scaleset_id)
+
+        logging.info("updating scaleset configs: %s", self.scaleset_id)
 
         pool = Pool.get_by_name(self.pool_name)
         if isinstance(pool, Error):
+            logging.error(
+                "unable to find pool during config update: %s - %s",
+                self.scaleset_id,
+                pool,
+            )
```

Comment:
i think needs_config_update  should be to set false here too

Reply:
In this case, we're marking the scaleset as failed, which moves it into the failed state.  If we ever have a scaleset without a pool, it's an _all-stop_ level event, as something is seriously broken.